Documentation/ApiOverview/LinkBrowser/Tutorials/CustomLinkBrowser.rst

```diff
@@ -0,0 +1,191 @@
+field with the old value).
+
+..  include:: /ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandlerCanHandleLink.rst.txt
+
+Format current URL
+------------------
+
+The function :php:`LinkHandlerInterface::formatCurrentUrl()` is used to preview
+what the link will look like in the backend, for example in the upper part of
+the link browser window.
+
+..  attention::
+    :php:`LinkHandlerInterface::formatCurrentUrl()` is not used to render the
+    link in the frontend.
+
+..  _tutorial-core-link-handler:
+
+2. Introduce the custom link format
+===================================
+
+You can find the complete class in the extension EXT:examples on GitHub:
+`GitHubLinkHandling <https://github.com/TYPO3-Documentation/t3docs-examples/blob/main/Classes/LinkHandler/GitHubLinkHandling.php>`__.
+
+Our :ref:`backend link handler implementation from step 1 <tutorial_backend_link_handler>`
+saves the link in the custom format `t3://github?issue=123` via JavaScript.
+
+This format is only an arbitrary string until we tell TYPO3 how to handle links
+of the new format by a second class which implements the
+:php:`TYPO3\CMS\Core\LinkHandling\LinkHandlingInterface`.
+
+..  note::
+    There are two interfaces with very similar names and very different
+    functionality involved here. The
+    :php:`\TYPO3\CMS\Backend\LinkHandler\LinkHandlerInterface` renders a tab in
+    the link browser window in the backend. Its implementing class is commonly
+    called a "(backend) link handler". Classes implementing the interface
+    :php:`TYPO3\CMS\Core\LinkHandling\LinkHandlingInterface` handle the
+    introduced link format. Such a class is called a "(core) link handler".
+
+..  include:: /ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GitHubLinkHandling.rst.txt
+
+The method :php:`LinkHandlingInterface::asString()` creates a string
+representation from the parameter array.
+
+:php:`LinkHandlingInterface::resolveHandlerData()` receives a parsed version of
+the string representation of the link and creates the parameter array from it.
+
+..  _tutorial-typolink-builder:
+
+3. Render the custom link format in the frontend
+================================================
+
+The link builder, a class extending the abstract class
+:php:`\TYPO3\CMS\Frontend\Typolink\AbstractTypolinkBuilder` is called whenever
+a link is rendered in the frontend, for example via
+TypoScript :typoscript:`.typolink`, by the
+:php:`\TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer::typoLink`
+function or by the :php:`\TYPO3\CMS\Extbase\Mvc\Web\Routing\UriBuilder`.
+
+..  include:: /ApiOverview/LinkBrowser/Tutorials/_CustomLinkBrowser/_GithubLinkBuilder.rst.txt
+
+The function :php:`AbstractTypolinkBuilder::build()` is called with the link
+configuration and data from the typolink function. If the link can be rendered
+it returns a new :php:`TYPO3\CMS\Frontend\Typolink\LinkResultInterface`. The
```

Comment:
```suggestion
it returns a new :php:`\TYPO3\CMS\Frontend\Typolink\LinkResultInterface`. The
```

Nope, is does not return an interface, it returns a class implementing the interface.